scripts/check_go_format.sh

```diff
@@ -23,20 +23,21 @@ emit_prunes() {
   for p in ${PRUNE_LIST}; do echo "-name ${p} -prune -o"; done | xargs
 }
 
-FILES=$(find ${ROOTS} $(emit_prunes) -name '*.go' -print | xargs)
+# shellcheck disable=SC2046
+FILES=$(find "${ROOTS[@]}" $(emit_prunes) -name '*.go' -print0 | xargs -0)
 
 if [ -z "$FILES" ]; then
     echo "no files to fix!" >&2
     exit 0
 fi
 
-if [ -n "${FIX}" ]; then
-  ${GOFMT} -s -w ${FILES}
+if [ -n "$FIX" ]; then
+  $GOFMT -s -w "$FILES"
```

Comment:
This is wrong. It will combine all the file names into a single argument. To do this correctly, I see two options: a) pipe the files into these `GOFMT` commands and use xargs. b) make `FILES` an array and construct it with [readarray](https://helpmanual.io/builtin/readarray/) on the results of find.

Reply:
My usual point when I think it is the time to stop using a shell script is when you need arrays or maps.

At that point, either (a) you prioritize transparency over generality ("yup, it doesn't handle paths with spaces in it, but if you want to know what the CI is doing it is running that command.") or (b) you notice that the users (and the execution environment) of script have expanded beyond the initial authoring group ("I ran this on RHEL4, it printed a bunch of stuff and nothing works").

If we think a particular script falls into (b), I'd suggest rewriting it in vanilla python, which doesn't have perfect dependency or version management, but will be more comprehensible to a larger set of future developers who will have to maintain it.

Reply:
This is a good point. What do you mean by "vanilla python" though? Do you mean Python using only the standard library? Which version of Python? I think defining the "shell + 1" language and environment would actually be quite useful.

Reply:
"vanilla python" being python using just the standard library and supporting all versions from last supported distro (nominally Ubuntu 18.04*) forward.

So python 3.6 I believe.

* I still consider supporting releases before 18.04 as best-effort, but if folks feel strongly about making earlier releases a hard requirement, that's fine too.

Reply:
I am 100% in favor of migrating this to python. Sounds like a separate PR though. Do linter scripts have to support pre 18.04 distros though? It seems reasonable to say we don't support developing on 16.04 directly.